### PR TITLE
Add top-level styles object

### PR DESCRIPTION
**Pull request by vincentmvdm**
Screenshot 2023-02-24 at 2 53 05 PM

## Changes 
- Adds a top-level `styles` key that includes extra information about all the styles in the dump, such as their key and name. Allows us to quickly load required styles and get extra info we can use to re-construct the user's DS
- You can control the inclusion of the styles object with the `styles` option. It defaults to false because you ideally only have to pull in this extra info about the user's DS (styles) every now and then. And unlike components, the style IDs already present on nodes are enough to use the styles in many cases
- Uses `figma-api-stub` for testing because my implementation uses `figma.getNodeById` among other things and this was the easiest way to test such functionality/create fake styles
    - I'd love for us to contribute back to the library at some point. It's the de-facto Figma testing library but could use some additions in a few areas
    
## Differences with Figma's REST API
- The styles object has the same structure as Figma's REST API but the ids used to store the styles is different than what Figma uses. We use the style's id whereas Figma uses a more arbitrary ID that's hard to predict. I've asked Figma for an explanation. Regardless, the functionality is the same.
- The Figma REST API replaces each node's `fillStyleId`, `strokeStyleId`, etc. with a single `styles` object that contains ids referring to the top-level styles object. I don't want to make that departure from the plugin API yet until we have a good reason (e.g. parity with REST API). The current behavior is backwards compatible and in line with how e.g. components have a `componentId` key
    
## Test plan
- Manual testing plus Jest tests

### CODE REVIEW

File: src/styles.test.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+  expect(d.styles).toEqual(styleMap);
+});
+
+test("Handles multiple styles of same type", async () => {
+  const { paintStyle } = createStyles();
+
+  const secondPaintStyle = figma.createPaintStyle();
+  // Hack because figma-api-stub doesn't create style keys.
+  (secondPaintStyle as { key: string }).key = getStyleKey(secondPaintStyle.id);
```
Comment (darknoon) [inline src/styles.test.ts:115]: Can you file an issue with `figma-api-stub` and link it here?

**Review by darknoon — APPROVED** [review]
👏 Great work!
Your explanation and tests leave nothing to the imagination and an easy +1 💚

**Comment by vincentmvdm** [comment]
> 👏 Great work! Your explanation and tests leave nothing to the imagination and an easy +1 💚

Thanks for the kind words :) Submitted an issue to `figma-api-stub`

File: src/styles.test.ts (hunk 1 above)
Comment (vincentmvdm) [reply src/styles.test.ts:115]: https://github.com/react-figma/figma-api-stub/issues/61

**Pull request merged**